src/core/components/start.js

```diff
@@ -34,6 +38,20 @@ module.exports = (self) => {
       },
       (cb) => self.libp2p.start(cb),
       (cb) => {
+        // Setup online routing for IPNS with a tiered routing composed by a DHT and a Pubsub router (if properly enabled)
+        const ipnsStores = []
+
+        // TODO Add IPNS pubsub if enabled
+
+        // NOTE: Until the IPNS over DHT is not ready, it is being replaced by the local repo datastore
+        // When DHT is added, If local option enabled, should receive offlineDatastore as well
+        const offlineDatastore = new OfflineDatastore(self._repo)
+        ipnsStores.push(offlineDatastore)
+
+        // Create ipns routing with a set of datastores
+        const routing = new TieredDatastore(ipnsStores)
+        self._ipns = new IPNS(routing, self)
```

Comment:
It would be easier to test and easier to understand what dependencies this module has if we just passed the things we need e.g. repo, keychain, peerId etc. instead of `self`